src/operator/nn/dnnl/dnnl_pooling-inl.h

```diff
@@ -172,27 +173,34 @@ inline bool DNNLRequireWorkspace(const PoolingParam& param) {
 }
 
 typedef ParamOpSign<PoolingParam> DNNLPoolingSignature;
-void DNNLPoolingCompute(const OpContext& ctx,
-                        const PoolingParam& param,
-                        const NDArray& in_data,
-                        const OpReqType req,
-                        const NDArray& out_data,
-                        const NDArray* workspace,
-                        const bool use_adaptive_pooling);
-
-void DNNLPoolingGradCompute(const OpContext& ctx,
-                            const PoolingParam& param,
-                            const NDArray& out_grad,
-                            const NDArray& in_data,
-                            const NDArray* workspace,
-                            const OpReqType req,
-                            const NDArray& in_grad);
+
+void DNNLPoolingGradCompute(const nnvm::NodeAttrs& attrs,
+                            const OpContext& ctx,
+                            const std::vector<NDArray>& inputs,
+                            const std::vector<OpReqType>& req,
+                            const std::vector<NDArray>& outputs);
 
 DNNLPoolingFwd& GetPoolingFwd(const PoolingParam& param,
                               const bool is_train,
                               const NDArray& data,
                               const NDArray& output,
                               const bool use_adaptive_pooling);
+
+template <bool use_adaptive_pooling>
+void DNNLPoolingCompute(const nnvm::NodeAttrs& attrs,
```

Comment:
Can this not stay in .cc file?

Reply:
So right now there are also some other definitions in this .h file 
#20776/files#diff-0c62b326967e9ddb0686baebd8cf8956ae723a5121e8f6a0fe5acfad70e2be6dR91-R92